### ISSUE TITLE
should use node 16, as node 12 is depreciated

### ISSUE THREAD

**Issue by mipmip** [samuelmeuli/action-electron-builder#87]
I get this message:

```
release (windows-latest)
Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: samuelmeuli/action-electron-builder
```

**Comment by christianlarrabure** [comment]
Can use this in the Install node step, right?

`        with:
          node-version: 16`

**Comment by msimic** [comment]
> Can use this in the Install node step, right?
> 
> ` with: node-version: 16`

no the problem is not on the github action you create with your project but inside this github action, it simply does not work anymore until it is updated to node 16

**Comment by petervanderwalt** [comment]
@samuelmeuli please please fix this (: 
We can't live without your Action! 

```
Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: actions/setup-node@v1, samuelmeuli/action-electron-builder@v1.6.0
```

**Comment by TechStudent10** [comment]
@samuelmeuli Yes, my project literally depends on this for releases. Please update it.

**Comment by njmulsqb** [comment]
> @samuelmeuli Yes, my project literally depends on this for releases. Please update it.

Just update the node version in the yml file from 10 to 16 and things will work for you, instead of waiting for the maintainer to fix it

**Comment by petervanderwalt** [comment]
> > @samuelmeuli Yes, my project literally depends on this for releases. Please update it.
> 
> Just update the node version in the yml file from 10 to 16 and things will work for you, instead of waiting for the maintainer to fix it

Is it https://github.com/samuelmeuli/action-electron-builder/blob/master/action.yml#L56 that should be changed?

**Comment by njmulsqb** [comment]
@petervanderwalt  No, in the yml file that you put in your `.github/workflows` directory, https://github.com/marketplace/actions/electron-builder-action

**Comment by petervanderwalt** [comment]
> @petervanderwalt No, in the yml file that you put in your `.github/workflows` directory, https://github.com/marketplace/actions/electron-builder-action

I have, see https://github.com/OpenBuilds/OpenBuilds-CONTROL/blob/55749cf07e6e4ebbacb59d5b5b25a493950660b0/.github/workflows/build.yml#L21 - but thats exactly when GA complains that the Action (which specifies https://github.com/samuelmeuli/action-electron-builder/blob/master/action.yml#L56 to use Node-12) results in 
`
Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: samuelmeuli/action-electron-builder
`

The "action" should be updated to use newer versions of Node - not the Workflow

**Comment by whutchinson98** [comment]
> @petervanderwalt No, in the yml file that you put in your `.github/workflows` directory, https://github.com/marketplace/actions/electron-builder-action

@njmulsqb It's not about the workflow you use it in. It's about the underlying dependencies that this action has (Node 12) which is why the action itself needs to be updated.

**Comment by petervanderwalt** [comment]
> why the action itself needs to be updated.

Exactly - So my earlier comment. 

> Is it https://github.com/samuelmeuli/action-electron-builder/blob/master/action.yml#L56 that should be changed?

After all then.

**Comment by whutchinson98** [comment]
> > why the action itself needs to be updated.
> 
> Exactly - So my earlier comment.
> 
> > Is it https://github.com/samuelmeuli/action-electron-builder/blob/master/action.yml#L56 that should be changed?
> 
> After all then.

Yep! I forked the repo and changed it myself and everything worked fine.

**Comment by njmulsqb** [comment]
@petervanderwalt Sorry, I am pretty new to github actions so couldnt understand your statement technically. I just changed the node version in my local workflows file and this action is now working fine with me and is producing releases perfectly.
@whutchinson98 How can one use your fork? I mean the current repo's action is available on marketplace and thus can be utilized from there. How can I use your fork in github actions? Thanks!

**Comment by whutchinson98** [comment]
> @petervanderwalt Sorry, I am pretty new to github actions so couldnt understand your statement technically. I just changed the node version in my local workflows file and this action is now working fine with me and is producing releases perfectly. @whutchinson98 How can one use your fork? I mean the current repo's action is available on marketplace and thus can be utilized from there. How can I use your fork in github actions? Thanks!

I have forked the repo into my companies organization. It's a public action and can be found [here](https://github.com/marketplace/actions/macro-electron-builder-action). Just a note that the readme is out of date as the action name has obviously changed. Simply click on the "use latest version" button to get the new action name `coparse-inc/action-electron-builder@v1.0.0`

**Comment by petervanderwalt** [comment]
> Yep! I forked the repo and changed it myself and everything worked fine.

Figured, thanks!

**Comment by TechStudent10** [comment]
> > @petervanderwalt Sorry, I am pretty new to github actions so couldnt understand your statement technically. I just changed the node version in my local workflows file and this action is now working fine with me and is producing releases perfectly. @whutchinson98 How can one use your fork? I mean the current repo's action is available on marketplace and thus can be utilized from there. How can I use your fork in github actions? Thanks!
> 
> I have forked the repo into my companies organization. It's a public action and can be found [here](https://github.com/marketplace/actions/macro-electron-builder-action). Just a note that the readme is out of date as the action name has obviously changed. Simply click on the "use latest version" button to get the new action name `coparse-inc/action-electron-builder@v1.0.0`

Thank you so much!

**Comment by vingeraycn** [comment]
Now you can use https://github.com/marketplace/actions/electron-builder-action-using-node-20 for node 20

> > @petervanderwalt Sorry, I am pretty new to github actions so couldnt understand your statement technically. I just changed the node version in my local workflows file and this action is now working fine with me and is producing releases perfectly. @whutchinson98 How can one use your fork? I mean the current repo's action is available on marketplace and thus can be utilized from there. How can I use your fork in github actions? Thanks!
> 
> I have forked the repo into my companies organization. It's a public action and can be found [here](https://github.com/marketplace/actions/macro-electron-builder-action). Just a note that the readme is out of date as the action name has obviously changed. Simply click on the "use latest version" button to get the new action name `coparse-inc/action-electron-builder@v1.0.0`

**Comment by petervanderwalt** [comment]
Thanks for the update, we forked at the time and figured it out. Also adapted the action to do the stuff we wanted, so we are good for now, but glad you updated it for everyone else (:
